src/java/org/apache/cassandra/dht/RangeStreamer.java

```diff
@@ -600,36 +601,26 @@ public StreamResultFuture fetchAsync()
                 // filter out already streamed ranges
                 RangesAtEndpoint available = stateStore.getAvailableRanges(keyspace, StorageService.instance.getTokenMetadata().partitioner);
 
-                //It's a bit unpredictable as to whether we need to fetch a replica or not
```

Comment:
Is there a reason we shouldn't explain what this predicate is doing?

Reply:
I thought I've added a more elaborate explanation than "it is unpredictable", but I'm happy to revise it.